File: Discrete Mathematics/Basics_Of_Digital_Logic_Designs/C++ Implementation/src/OR_GATE.cpp
#include "OR_GATE.h"

int OR_GATE::is_OR_GATE(int x, int y)
{
    if ((x && y)<=1 && (x&&y)>=0)
    {
        if ((x||y)==1)
        {
            return 1;
        }
        else
        {
            return 0;
        }
    }
    else
    {
        cout<< "Inputs must be 0 or 1 \n";
    }
}

File: Discrete_Mathematics/Basics_Of_Digital_Logic_Designs/Java_Implementation/OR_gate.java
// Java Implementation of AND gate

public class OR_gate {

    public String is_OR_gate(int p, int q){

        if ((p == 0 || p == 1)&&(q == 0||q==1)){

            if (p == 1 || q==1){
                return "1";
            }
            else{
                return "0";
            }
        }

        else{
            System.out.println("Invalid input \n Input must be 0 or 1...");
        }
        return "";
    }


    // Test
    String output;
    public void main(String[] args) {
        output =  is_OR_gate(1,1);
        System.out.println(output); // output: 1

        output = is_OR_gate(0, 0);
        System.out.println(output); // output: 0

        output = is_OR_gate(1, 0);
        System.out.println(output); // output: 1

        output = is_OR_gate(0, 1);
        System.out.println(output); // output: 1
    }
}
